catalyst_voices/lib/pages/account/delete_keychain_dialog.dart

```diff
@@ -0,0 +1,156 @@
+                  Wrap(
+                    children: [
+                      VoicesAssets.icons.exclamation.buildIcon(
+                        color: Theme.of(context).colors.iconsError,
+                      ),
+                      Padding(
+                        padding: const EdgeInsets.only(top: 2, left: 8),
+                        child: Text(
+                          '''
+Make sure you have a working Catalyst 12-word seedphrase!''',
+                          style: Theme.of(context).textTheme.titleSmall,
+                        ),
+                      ),
+                    ],
+                  ),
+                  const SizedBox(height: 8),
+                  Text(
+                    '''
+Your Catalyst account will be removed, this action cannot be undone!''',
+                    style: Theme.of(context).textTheme.bodyMedium,
+                    textAlign: TextAlign.center,
+                  ),
+                  const SizedBox(height: 48),
+                  Text(
+                    'To avoid mistakes, please type ‘Remove Keychain’ below.',
+                    style: Theme.of(context).textTheme.titleSmall,
+                  ),
+                  const SizedBox(height: 24),
+                  Wrap(
+                    direction: Axis.vertical,
+                    children: [
+                      Text(
+                        'Confirm removal',
+                        style: Theme.of(context).textTheme.titleSmall,
+                      ),
+                      const SizedBox(height: 2),
+                      SizedBox(
+                        width: 300,
+                        child: VoicesTextField(
+                          controller: _textEditingController,
+                          decoration: VoicesTextFieldDecoration(
+                            errorText: _errorText,
+                            filled: true,
+                            fillColor: Theme.of(context)
+                                .colors
+                                .elevationsOnSurfaceNeutralLv1White,
+                          ),
+                        ),
+                      ),
+                    ],
+                  ),
+                  const SizedBox(height: 24),
+                  Wrap(
+                    children: [
+                      VoicesFilledButton(
+                        backgroundColor: Theme.of(context).colors.iconsError,
+                        onTap: () async {
+                          if (_textEditingController.text ==
+                              'Remove Keychain') {
+                            // TODO(Jakub): remove keychain
+                            Navigator.of(context).pop();
+                            await VoicesDialog.show<void>(
+                              context: context,
+                              builder: (context) {
+                                return const KeychainDeletedDialog();
+                              },
+                            );
```

Comment:
Does it make sense for this dialog to open another one? Maybe it should be better for this to, lets say, return "true" with pop `Navigator.of(context).pop(true)` 

```dart
final confirmed = await DeleteKeychainConfirmationDialog.show(context);

if(confirmed && mounted){
     KeychainDeletedDialog.show(context);
}

```

Reply:
I find passing value like that messy, however, I agree that dialog shouldn't open another one so I've introduced `onDeleteKeychain` callback

Reply:
Returning value from dialog, it's quite common, i.e. `showDatePicker()` from flutter has return type of `Future<DateTime?>`.

But lets do what Damian suggested, the messy part is that `showDialog` method has generic return type and will accept any type. To make it not messy we need a wrapper that will show the dialog for us and return expected type. You can see that on `DeleteKeychainConfirmationDialog.show(context);`

You would define it as:

```dart
class DeleteKeychainConfirmationDialog {
  static Future<bool?> show(BuildContext context) {
    // show dialog and cast the type
  }
}
```

Reply:
Its also every easy to do smth like this

```dart
class DeleteKeychainConfirmationDialog {
  static Future<bool> show(BuildContext context) async {
    final result =await  VoicesDialog.show<bool>(
      context: context,
      builder: (context) => ...,
    );
    
    return result ?? false;
  }
}
```

Reply:
It's not fully clear for me. Would you like every dialog to have its own static `show` function?
It's simple `bool` type so what we need to map it to? Do you mean that this function will have named bool and that's why it's gonna be more semantic?